server/kube_health_test.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package protostruct supports operations on the protocol buffer Struct message.
+package server
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/apigee/apigee-remote-service-golib/product"
+	"google.golang.org/grpc/health"
+	"google.golang.org/grpc/health/grpc_health_v1"
+)
+
+func TestKubeHealth(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var result = product.APIResponse{
+			APIProducts: []product.APIProduct{},
+		}
+		w.Header().Set("Content-Type", "application/json")
+		_ = json.NewEncoder(w).Encode(result)
+	}))
+	defer ts.Close()
+
+	serverURL, err := url.Parse(ts.URL)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	productMan, err := product.NewManager(product.Options{
+		Client:      http.DefaultClient,
+		BaseURL:     serverURL,
+		RefreshRate: time.Minute,
+		Org:         "org",
+		Env:         "env",
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer productMan.Close()
+
+	grpcHealth := health.NewServer()
+	handler := &Handler{
+		productMan: productMan,
+	}
+
+	kubeHealth := NewKubeHealth(handler, grpcHealth)
+	err = kubeHealth.error()
```

Comment:
If I didn't miss anything, this is not guaranteed to be called before the product is loaded. While it probably works in normal test condition, it theoretically could fail? Would it make sense to block the product loading in a more robust way till the error is unwanted?

Reply:
Sure. I got lazy. I'll guarantee it fails. (That said, guaranteeing the next test succeeds would be harder.)